lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -537,12 +537,22 @@ def create_transaction_parameters(money, credit_card_or_vault_id, options)
             :first_name => credit_card_or_vault_id.first_name,
             :last_name => credit_card_or_vault_id.last_name
           )
-          parameters[:credit_card] = {
-            :number => credit_card_or_vault_id.number,
-            :cvv => credit_card_or_vault_id.verification_value,
-            :expiration_month => credit_card_or_vault_id.month.to_s.rjust(2, "0"),
-            :expiration_year => credit_card_or_vault_id.year.to_s
-          }
+          if credit_card_or_vault_id.is_a?(NetworkTokenizationCreditCard)
```

Comment:
@jnormore I've seen something like this `is_a?` check in all the adapters so far. Wondering if it shouldn't be the responsibility of the credit card/payment method to know what its type is? Something like:

```ruby
payment_method.tokenized?
```

Also, I feel like we need to decide if we're going to be generic and call this type of support "network tokenization" support or be specific and call it "Apple Pay" support. Some gateways (Cybersource) seem to implement true generic network tokenization support whereas others, like Braintree here, are specific to Apple Pay as evidence by the `apple_pay_card` parameter.

You've seen more of these than I, how are gateways implementing this feature – In true generic fashion, or specific to AP?